### ISSUE TITLE
Database doesn't work when created from larger amount of data on versions 0.3.4 and 0.4.0

### ISSUE THREAD

**Issue by FelipeLagare** [duckdb/duckdb#3949]
#### What happens?
We've been using duckdb version 0.3.2 to create and use our database, but when using the same data to create the schemas with versions 0.3.4 and 0.4.0 the database does not work, as it is not possible to query data neither pragma queries show anything, but the file exists and sizes the expected amount.

#### To Reproduce
We are running one python script with one connection and looping to create the schemas, tables and populate them with insert statements:

CREATE SCHEMA IF NOT EXISTS {schema}
CREATE TABLE {schema}.{table} (
            {columns})
INSERT INTO {schema}.{table}
SELECT {columns}
FROM read_csv_auto({path}, columns={column}, header=False, sample_size=-1)

Apparently, a possible cause to this issue is the amount of data. I've tried to run the script with fewer csv files and it worked fine.
Important to notice we're also splitting larger csv files in chunks: `pd.read_csv(path, header=None, chunksize=1000000)`.

#### Environment (please complete the following information):
 - OS: Windows 10
 - DuckDB Version: 0.3.4 and 0.4.0
 - DuckDB Client: Python 3.8.8

#### Identity Disclosure:
 - Full Name: Felipe Lagares
 - Affiliation: Admolly

#### Before Submitting

- [ ] **Have you tried this on the latest `master` branch?**
Yes.

**Comment by hannes** [comment]
Can you try to make a reproducible example please?

**Comment by djouallah** [comment]
I think it is a duplicate of this bug report
https://github.com/duckdb/duckdb/issues/3438

**Comment by FelipeLagare** [comment]
Yes, I think my issue is similar to the bug report mentioned above, except that I'm inserting directly from csv files.
Also, the breaking point seems to be something between 2Gb and 2.5Gb.

Thanks @djouallah
